src/mechanics/constitutive/laws/MechanicsInterface.h

```diff
@@ -11,8 +11,9 @@ template <int TDim>
 struct MechanicsInterface
 {
     using MechanicsTangent = Eigen::Matrix<double, Voigt::Dim(TDim), Voigt::Dim(TDim)>;
-    virtual EngineeringStressPDE<TDim> Stress(EngineeringStrainPDE<TDim> strain, double deltaT) const = 0;
-    virtual MechanicsTangent Tangent(EngineeringStrainPDE<TDim>, double deltaT) const = 0;
+    virtual EngineeringStressPDE<TDim> Stress(EngineeringStrainPDE<TDim> strain, double deltaT, int cellNum,
```

Comment:
Naming consistency: `cellNum = cellData.GetCellId()`?

Reply:
will change that

Reply:
provide already the entry in the vector  cellNum*numIpPerCell+ipNum